test/xpu_api/random/unit_tests/philox_unit_test.pass.cpp

```diff
@@ -338,42 +362,99 @@ template <typename Engine>
 int
 discard_overflow_test()
 {
+    int ret_sts = 0;
+
     using T = typename Engine::result_type;
     using scalar_type = typename Engine::scalar_type;
 
-    Engine engine1;
-    std::array<T, Engine::word_count> counter;
-
-    for (int i = 0; i < Engine::word_count; i++)
+    // Iterate through the counter's position being overflown
+    for (int overflown_position = 0; overflown_position < Engine::word_count - 1; ++overflown_position)
     {
-        counter[i] = 0;
-    }
+        Engine engine1;
+        std::array<T, Engine::word_count> counter = {0};
 
-    // overflow of the 0th counter element
-    counter[2] = 1;
+        // Overflow of a counter's element. The correspondence is the following:
+        // 0 1 2 3 overflow position
+        // 1 2 3 0 set-up counter position in engine
+        // 2 1 0 3 raw_counter_position
+        int raw_counter_position = (Engine::word_count - overflown_position - 2) % Engine::word_count;
+        counter[raw_counter_position] = 1;
 
-    engine1.set_counter(counter);
+        engine1.set_counter(counter);
 
-    Engine engine2;
+        Engine engine2;
 
-    for (int i = 0; i < Engine::word_count; i++)
-    {
-        engine2();
+        // To reduce the execution time pre-set counter to almost-overflown state
+        std::array<T, Engine::word_count> counter2 = {0};
+        for (int i = Engine::word_count - overflown_position - 1; i < Engine::word_count - 1; ++i)
+        {
+            counter2[i] = std::numeric_limits<unsigned long long>::max();
+        }
+
+        engine2.set_counter(counter2);
+
+        for (int i = 0; i < Engine::word_count; i++)
+        {
+            engine2();
+        }
+
+        for (int i = 0; i < Engine::word_count; i++)
+        {
+            engine2.discard(engine2.max());
+        }
+
+        if (engine1() == engine2())
+        {
+            ret_sts = 0;
+        }
+        else
+        {
+            std::cout << " failed" << std::endl;
+            ret_sts = 1;
+            break;
+        }
     }
-    for (int i = 0; i < Engine::word_count; i++)
+
+    if (!ret_sts)
+        std::cout << " passed" << std::endl;
+
+    return ret_sts;
+}
+
+template <typename Engine>
+int
+counter_management_test()
+{
+    using T = typename Engine::result_type;
+    Engine testedEngine;
+    Engine referenceEngine;
+
+    // set the counter which value is 2-chunk bitsize
+    unsigned long long increment = ((unsigned long long)testedEngine.max() << Engine::word_size) | testedEngine.max();
+    unsigned long long counter_increment = increment / 4;
```

Comment:
```suggestion
    unsigned long long counter_increment = increment / Engine::word_count;
```

Reply:
Thanks for catching that! Fixed + all unit tests are also extended for Philox 2x*(except for `set_counter_conformance_test` as it expects `1955073260` and `3409172418970261260` numbers to be generated as 10000th)